timemachine/fe/free_energy.py

```diff
@@ -786,6 +786,21 @@ def run_sims_hrex(
         HREX statistics (e.g. swap rates, replica-state distribution)
     """
 
+    # TODO: to support replica exchange with variable temperatures,
+    #  consider modifying sample fxn to rescale velocities by sqrt(T_new/T_orig)
+    def _assert_velocity_distributions_compatible(state_a: InitialState, state_b: InitialState):
+        """assert masses and temperature match.
+        (so that -- up to timestep error -- (x, v, b) in state_a can be swapped with (x, v, b) from state_b)"""
+
+        intg_a = state_a.integrator
+        intg_b = state_b.integrator
+
+        assert (intg_a.masses == intg_b.masses).all()
+        assert intg_a.temperature == intg_b.temperature
```

Comment:
question: should we also assert that pressures are equal?

Reply:
Ahh, good call, was waffling about that.

(We seem less likely to ever change pressures, but would be good to enforce that assumption. If the pressures were unequal, then the `u_kln` matrix would just have to be computed differently (including [a `pressure * volume` term ](https://github.com/proteneer/timemachine/blob/7ac5ff6f50ef6d58e7f11b4cebe7beb2b3487730/attic/modules/md/ensembles.py#L119-L122) ).)